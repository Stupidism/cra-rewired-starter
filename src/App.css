.App {
  height: 100vh;
  position: relative;
}

.App .header {
  position: absolute;
  top: 0;
  left: 0;
  width: 200px;
  padding: 0;
  z-index: 10000;
  color: white;
}

.App .trigger {
  color: white;
  font-size: 18px;
  line-height: 64px;
  padding: 0 23px;
  cursor: pointer;
  transition: color 0.3s;
}

.App .trigger:hover {
  color: #108ee9;
}

.App .logo {
  height: 32px;
  background: #333;
  border-radius: 6px;
  margin: 16px;
}

.App .sider {
  padding-top: 64px;
}
