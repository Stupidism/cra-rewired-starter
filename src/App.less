.App {
  height: 100vh;
  position: relative;

  .header {
    position: absolute;
    top: 0;
    left: 0;
    width: 200px;
    padding: 0;
    z-index: 10000;
    color: white;
    font-size: 16px;
    font-weight: bold;
  }

  .trigger {
    color: white;
    font-size: 18px;
    line-height: 64px;
    padding: 0 23px;
    cursor: pointer;
    transition: color 0.3s;

    &:hover {
      color: #108ee9;
    }
  }

  .sider {
    padding-top: 64px;
  }
}
