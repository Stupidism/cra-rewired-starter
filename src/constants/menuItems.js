export default [
  {
    name: 'Current Status',
    icon: 'clock-circle-o',
  },
  {
    name: 'Historical Data',
    icon: 'calendar',
  },
  {
    name: 'Node List',
    icon: 'bars',
  },
  {
    name: 'Group List',
    icon: 'api',
  },
  {
    name: 'Tag List',
    icon: 'tags-o',
  },
  {
    name: 'Settings',
    icon: 'setting',
  },
  {
    name: 'Sign Out',
    icon: 'logout',
  },
];
