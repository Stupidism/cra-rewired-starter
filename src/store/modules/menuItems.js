import { createReducer } from 'redux-create-reducer';

// ------------------------------------
// Constants
// ------------------------------------
export const SELECT_MENU_ITEM = 'SELECT_MENU_ITEM';
export const menuItems = [
  {
    name: 'Current Status',
    icon: 'clock-circle-o',
  },
  {
    name: 'Historical Data',
    icon: 'calendar',
  },
  {
    name: 'Node List',
    icon: 'bars',
  },
  {
    name: 'Group List',
    icon: 'api',
  },
  {
    name: 'Tag List',
    icon: 'tags-o',
  },
  {
    name: 'Settings',
    icon: 'setting',
  },
  {
    name: 'Sign Out',
    icon: 'logout',
  },
];

const defaultState = {
  selected: 0,
};

// ------------------------------------
// Actions
// ------------------------------------
export const selectMenuItem = value => ({ type: SELECT_MENU_ITEM, value });

// ------------------------------------
// Reducers
// ------------------------------------
export default createReducer(defaultState, {
  [SELECT_MENU_ITEM]: (state, { value }) => ({ selected: value }),
});
