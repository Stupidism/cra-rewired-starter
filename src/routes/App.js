import React, { Component } from 'react';
import { Route, Switch } from 'react-router-dom';

import Header from 'containers/Header';

import HomePage from './HomePage';
import NotFoundPage from './NotFoundPage';

import './App.less';

class App extends Component {
  render() {
    return (
      <div className="App">
        <Header />
        <Switch>
          <Route exact path="/" component={HomePage} />
          <Route component={NotFoundPage} />
        </Switch>
      </div>
    );
  }
}

export default App;
