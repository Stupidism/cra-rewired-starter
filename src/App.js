// @flow
import * as React from 'react';
import { Layout, Menu, Icon } from 'antd';
import { connect } from 'react-redux';
import { bindActionCreators } from 'redux';

import { menuItems, selectMenuItem } from './store/modules/menuItems';

import './App.css';

const { Header, Sider, Content } = Layout;

// ------------------------------------
// Types
// ------------------------------------
type OnSelectMenuItem = ({ key: string }) => mixed;

type Props = {
  onSelectMenuItem: OnSelectMenuItem,
  selectedMenuItem: { name: string, icon: string },
};

type State = {
  collapsed: boolean,
};

// ------------------------------------
// Component
// ------------------------------------
class SiderDemo extends React.Component<Props, State> {
  state = {
    collapsed: false,
  };

  toggle = () => {
    this.setState({
      collapsed: !this.state.collapsed,
    });
  };

  render() {
    return (
      <Layout className="App">
        <Header className="header">
          <Icon
            className="trigger"
            type={this.state.collapsed ? 'menu-unfold' : 'menu-fold'}
            onClick={this.toggle}
          />
          {this.props.selectedMenuItem.name}
        </Header>

        <Sider
          className="sider"
          trigger={null}
          collapsible
          collapsedWidth={0}
          collapsed={this.state.collapsed}
        >
          <Menu
            theme="dark"
            mode="inline"
            defaultSelectedKeys={['0']}
            onSelect={this.props.onSelectMenuItem}
          >
            {menuItems.map(({ name, icon }, index) => (
              <Menu.Item key={index}>
                <Icon type={icon} />
                <span>{name}</span>
              </Menu.Item>
            ))}
          </Menu>
        </Sider>
        <Content
          style={{
            margin: '24px 16px',
            padding: 24,
            background: '#fff',
            minHeight: 280,
          }}
        >
          Content
        </Content>
      </Layout>
    );
  }
}

// ------------------------------------
// Handlers
// ------------------------------------
const onSelectMenuItem: OnSelectMenuItem = ({ key }) =>
  selectMenuItem(Number(key));

const mapStateToProps = state => ({
  selectedMenuItem: menuItems[state.menuItems.selected],
});

const mapDispatchToProps = dispatch =>
  bindActionCreators(
    {
      onSelectMenuItem,
    },
    dispatch,
  );

export default connect(mapStateToProps, mapDispatchToProps)(SiderDemo);
