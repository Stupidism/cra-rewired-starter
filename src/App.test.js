import React from 'react';
import ReactDOM from 'react-dom';
import renderer from 'react-test-renderer';
import menuItems from 'constants/menuItems';

import { App } from './App';

describe('<App />', () => {
  it('renders without crashing', () => {
    const div = document.createElement('div');
    ReactDOM.render(<App selectedMenuItem={menuItems[0]} />, div);
  });

  it('keeps unchanged', () => {
    const tree = renderer
      .create(<App selectedMenuItem={menuItems[1]} />)
      .toJSON();

    expect(tree).toMatchSnapshot();
  });
});
