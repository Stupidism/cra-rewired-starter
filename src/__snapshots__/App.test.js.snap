// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<App /> keeps child components unchanged 1`] = `
<div
  className="App ant-layout"
>
  <Adapter
    className="header"
  >
    <Icon
      className="trigger"
      onClick={[Function]}
      type="menu-fold"
    />
    Historical Data
  </Adapter>
  <Sider
    className="sider"
    collapsed={false}
    collapsedWidth={80}
    collapsible={true}
    defaultCollapsed={false}
    prefixCls="ant-layout-sider"
    reverseArrow={false}
    style={Object {}}
    trigger={null}
    width={200}
  >
    <Menu
      className=""
      defaultSelectedKeys={
        Array [
          "0",
        ]
      }
      focusable={false}
      mode="inline"
      prefixCls="ant-menu"
      theme="dark"
    >
      <MenuItem
        key="0"
      >
        <Icon
          type="clock-circle-o"
        />
        <span>
          Current Status
        </span>
      </MenuItem>
      <MenuItem
        key="1"
      >
        <Icon
          type="calendar"
        />
        <span>
          Historical Data
        </span>
      </MenuItem>
      <MenuItem
        key="2"
      >
        <Icon
          type="bars"
        />
        <span>
          Node List
        </span>
      </MenuItem>
      <MenuItem
        key="3"
      >
        <Icon
          type="api"
        />
        <span>
          Group List
        </span>
      </MenuItem>
      <MenuItem
        key="4"
      >
        <Icon
          type="tags-o"
        />
        <span>
          Tag List
        </span>
      </MenuItem>
      <MenuItem
        key="5"
      >
        <Icon
          type="setting"
        />
        <span>
          Settings
        </span>
      </MenuItem>
      <MenuItem
        key="6"
      >
        <Icon
          type="logout"
        />
        <span>
          Sign Out
        </span>
      </MenuItem>
    </Menu>
  </Sider>
  <Adapter
    style={
      Object {
        "background": "#fff",
        "margin": "24px 16px",
        "minHeight": 280,
        "padding": 24,
      }
    }
  >
    Content
  </Adapter>
</div>
`;

exports[`<App /> keeps unchanged in normal state 1`] = `
<div
  className="App ant-layout ant-layout-has-sider"
>
  <div
    className="header ant-layout-header"
  >
    <i
      className="anticon anticon-menu-fold trigger"
      onClick={[Function]}
    />
    Historical Data
  </div>
  <div
    className="sider ant-layout-sider"
    style={
      Object {
        "flex": "0 0 200px",
        "maxWidth": "200px",
        "minWidth": "200px",
        "width": "200px",
      }
    }
  >
    <div
      className="ant-layout-sider-children"
    >
      <ul
        className="ant-menu ant-menu-dark ant-menu-root ant-menu-inline"
        role="menu"
        style={Object {}}
      >
        <li
          aria-disabled={undefined}
          className="ant-menu-item ant-menu-item-selected"
          onBlur={undefined}
          onClick={[Function]}
          onContextMenu={undefined}
          onFocus={undefined}
          onMouseDown={undefined}
          onMouseEnter={[Function]}
          onMouseLeave={[Function]}
          onTouchStart={undefined}
          role="menuitem"
          style={
            Object {
              "paddingLeft": 24,
            }
          }
          title={undefined}
        >
          <i
            className="anticon anticon-clock-circle-o"
          />
          <span>
            Current Status
          </span>
        </li>
        <li
          aria-disabled={undefined}
          className="ant-menu-item"
          onBlur={undefined}
          onClick={[Function]}
          onContextMenu={undefined}
          onFocus={undefined}
          onMouseDown={undefined}
          onMouseEnter={[Function]}
          onMouseLeave={[Function]}
          onTouchStart={undefined}
          role="menuitem"
          style={
            Object {
              "paddingLeft": 24,
            }
          }
          title={undefined}
        >
          <i
            className="anticon anticon-calendar"
          />
          <span>
            Historical Data
          </span>
        </li>
        <li
          aria-disabled={undefined}
          className="ant-menu-item"
          onBlur={undefined}
          onClick={[Function]}
          onContextMenu={undefined}
          onFocus={undefined}
          onMouseDown={undefined}
          onMouseEnter={[Function]}
          onMouseLeave={[Function]}
          onTouchStart={undefined}
          role="menuitem"
          style={
            Object {
              "paddingLeft": 24,
            }
          }
          title={undefined}
        >
          <i
            className="anticon anticon-bars"
          />
          <span>
            Node List
          </span>
        </li>
        <li
          aria-disabled={undefined}
          className="ant-menu-item"
          onBlur={undefined}
          onClick={[Function]}
          onContextMenu={undefined}
          onFocus={undefined}
          onMouseDown={undefined}
          onMouseEnter={[Function]}
          onMouseLeave={[Function]}
          onTouchStart={undefined}
          role="menuitem"
          style={
            Object {
              "paddingLeft": 24,
            }
          }
          title={undefined}
        >
          <i
            className="anticon anticon-api"
          />
          <span>
            Group List
          </span>
        </li>
        <li
          aria-disabled={undefined}
          className="ant-menu-item"
          onBlur={undefined}
          onClick={[Function]}
          onContextMenu={undefined}
          onFocus={undefined}
          onMouseDown={undefined}
          onMouseEnter={[Function]}
          onMouseLeave={[Function]}
          onTouchStart={undefined}
          role="menuitem"
          style={
            Object {
              "paddingLeft": 24,
            }
          }
          title={undefined}
        >
          <i
            className="anticon anticon-tags-o"
          />
          <span>
            Tag List
          </span>
        </li>
        <li
          aria-disabled={undefined}
          className="ant-menu-item"
          onBlur={undefined}
          onClick={[Function]}
          onContextMenu={undefined}
          onFocus={undefined}
          onMouseDown={undefined}
          onMouseEnter={[Function]}
          onMouseLeave={[Function]}
          onTouchStart={undefined}
          role="menuitem"
          style={
            Object {
              "paddingLeft": 24,
            }
          }
          title={undefined}
        >
          <i
            className="anticon anticon-setting"
          />
          <span>
            Settings
          </span>
        </li>
        <li
          aria-disabled={undefined}
          className="ant-menu-item"
          onBlur={undefined}
          onClick={[Function]}
          onContextMenu={undefined}
          onFocus={undefined}
          onMouseDown={undefined}
          onMouseEnter={[Function]}
          onMouseLeave={[Function]}
          onTouchStart={undefined}
          role="menuitem"
          style={
            Object {
              "paddingLeft": 24,
            }
          }
          title={undefined}
        >
          <i
            className="anticon anticon-logout"
          />
          <span>
            Sign Out
          </span>
        </li>
      </ul>
    </div>
  </div>
  <div
    className="ant-layout-content"
    style={
      Object {
        "background": "#fff",
        "margin": "24px 16px",
        "minHeight": 280,
        "padding": 24,
      }
    }
  >
    Content
  </div>
</div>
`;

exports[`<App /> keeps unchanged when menu is collapsed 1`] = `
<div
  className="App ant-layout ant-layout-has-sider"
>
  <div
    className="header ant-layout-header"
  >
    <i
      className="anticon anticon-menu-unfold trigger"
      onClick={[Function]}
    />
    Historical Data
  </div>
  <div
    className="sider ant-layout-sider ant-layout-sider-collapsed"
    style={
      Object {
        "flex": "0 0 80px",
        "maxWidth": "80px",
        "minWidth": "80px",
        "width": "80px",
      }
    }
  >
    <div
      className="ant-layout-sider-children"
    >
      <ul
        className="ant-menu ant-menu-dark ant-menu-inline-collapsed ant-menu-root ant-menu-vertical"
        role="menu"
        style={Object {}}
      >
        <li
          className="ant-menu-item ant-menu-item-selected"
          onClick={[Function]}
          onMouseEnter={[Function]}
          onMouseLeave={[Function]}
          role="menuitem"
          style={Object {}}
        >
          <i
            className="anticon anticon-clock-circle-o"
          />
          <span>
            Current Status
          </span>
        </li>
        <li
          className="ant-menu-item"
          onClick={[Function]}
          onMouseEnter={[Function]}
          onMouseLeave={[Function]}
          role="menuitem"
          style={Object {}}
        >
          <i
            className="anticon anticon-calendar"
          />
          <span>
            Historical Data
          </span>
        </li>
        <li
          className="ant-menu-item"
          onClick={[Function]}
          onMouseEnter={[Function]}
          onMouseLeave={[Function]}
          role="menuitem"
          style={Object {}}
        >
          <i
            className="anticon anticon-bars"
          />
          <span>
            Node List
          </span>
        </li>
        <li
          className="ant-menu-item"
          onClick={[Function]}
          onMouseEnter={[Function]}
          onMouseLeave={[Function]}
          role="menuitem"
          style={Object {}}
        >
          <i
            className="anticon anticon-api"
          />
          <span>
            Group List
          </span>
        </li>
        <li
          className="ant-menu-item"
          onClick={[Function]}
          onMouseEnter={[Function]}
          onMouseLeave={[Function]}
          role="menuitem"
          style={Object {}}
        >
          <i
            className="anticon anticon-tags-o"
          />
          <span>
            Tag List
          </span>
        </li>
        <li
          className="ant-menu-item"
          onClick={[Function]}
          onMouseEnter={[Function]}
          onMouseLeave={[Function]}
          role="menuitem"
          style={Object {}}
        >
          <i
            className="anticon anticon-setting"
          />
          <span>
            Settings
          </span>
        </li>
        <li
          className="ant-menu-item"
          onClick={[Function]}
          onMouseEnter={[Function]}
          onMouseLeave={[Function]}
          role="menuitem"
          style={Object {}}
        >
          <i
            className="anticon anticon-logout"
          />
          <span>
            Sign Out
          </span>
        </li>
      </ul>
    </div>
  </div>
  <div
    className="ant-layout-content"
    style={
      Object {
        "background": "#fff",
        "margin": "24px 16px",
        "minHeight": 280,
        "padding": 24,
      }
    }
  >
    Content
  </div>
</div>
`;
